### DISCUSSION TITLE
Prompt the user when another application modifies the currently opened file

### DISCUSSION THREAD

**Discussion by hexagonrecursion** [gsantner/markor#1607] [Ideas]
It is not uncommon for the currently opened file to be changed by another application. Maybe it was edited in a different editor. Maybe a newer version was synced from another machine. Most test editors I have used on Linux prompt me "The file on disk changed. Reload? (Reload, Keep this version)"

It appears that this feature is not currently implemented in Markor.

**Comment by gsantner** [comment]
Markor automatically saves the file when you change to other app. And when you come back to markor, it is automatically reloaded (in case there was a change since markors current version).

So there is no need to ask, all is done automatically for you.

**Reply by hexagonrecursion** [reply]
What if:
1. I modify a file on my computer
2. Switch to my phone
3. Open markor
4. Syncthing syncs my updates

**Reply by gsantner** [reply]
there are endless possibilities, but time of developers is not unlimited.

most likely you get a sync conflict by syncthing. that is what happens to me when I edit somewhat on nearly same time on multiple devices.

otherwise - if you look for a collaborative editor where multiple people work on the same thing at the same time - there is e.g. Etherpad. And as always - have backups. Syncthing has that (e.g. last 3 versions) builtin.

Markor is there to help you with many things, but it can't make everybody happy. 

Don't see the need here to add additional logic, risk breaking things, have annoying popups ... all just to fit for a very specific edge case :-).

**Reply by hexagonrecursion** [reply]
> most likely you get a sync conflict by syncthing

Not always. Not reliably. I just tested:

1. I modify a file on my computer
2. Switch to my phone
3. Open markor
4. Syncthing syncs my updates
5. I do not notice that I am not editing the latest version
6. Edit&save in markor
7. Syncthing syncs back to my computer and changes are lost

**Reply by hexagonrecursion** [reply]
> And as always - have backups

My current backup setup would be useless here. How would you implement a backup that would save me in this situation?

**Reply by harshad1** [reply]
Syncthing has the ability to save the last n versions of every file (the simple backup scheme). This is what I use.

**Comment by harshad1** [comment]
@gsantner Continuously checking if a file has changed in the background is problematic, but it may be possible to detect conflicts at save time and save a 'conflict' file...
